pkg/policy/l3.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2016-2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policy
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"net"
+	"strconv"
+	//	"github.com/cilium/cilium/api/v1/models"
+
+	"github.com/cilium/cilium/pkg/maps/cidrmap"
+)
+
+// L3PolicyMap is a list of CIDR filters indexable by address/prefixlen
+// key format: "address/prefixlen", e.g., "10.1.1.0/24"
+type L3PolicyMap struct {
+	Map         map[string]net.IPNet // Allowed L3 prefixes
```

Comment:
Regeneration currently always creates a new L3Policy and actually does not compare the new one to old one. BPF regeneration uses the endpoint.L3Policy, but that is also run from the same thread. endpoint.DeepCopy() could potentially be called from another thread in parallel, but it is only used by the unit tests.

In summary, it seems to me that each L3Policy is accessed from one thread only and further locking is not needed.